src/skdh/context/motion_detect.py

```diff
@@ -38,42 +58,301 @@ def __init__(self, filter_cutoff=6, cov_threshold=0.022558169349678834):
+        time : numpy.ndarray
+            (N, ) array of unix timestamps, in seconds.
         accel : numpy.ndarray
-            (N, 3) array of acceleration, in units of 'g', collected at 20hz.
+            (N, 3) array of acceleration, in units of 'g'.
         fs : float
-            Sampling frequency.
+            Sampling rate. Default None. If not provided, will be inferred.
 
         Returns
         -------
         results : dict
-            Results dictionary including detected motion (1s rolling) and raw values of rolling CoV.
+            Results dictionary including detected motion (1s rolling) and raw values of 1s rolling CoV.
 
         """
+        # check input requirements are met
+        time, accel, fs = self._check_input(time, accel)
+
         # Vector Magnitude
         vmag = np.linalg.norm(accel, axis=1)
 
         # Low-pass filter the accelerometer vector magnitude signal to remove high frequency components
+        # Note: iirfilter with filtfilt used to match legacy implementation
         wn = self.filter_cutoff * 2 / fs
         [b, a] = iirfilter(self.filter_cutoff, wn, btype="lowpass", ftype="butter")
         vmag_filt = filtfilt(b, a, vmag)
 
         # Calculate the 1s rolling coefficient of variation
-        rolling_mean = moving_mean(a=vmag_filt, w_len=int(fs), skip=1)
-        rolling_std = moving_sd(a=vmag_filt, w_len=int(fs), skip=1)
-        rolling_cov = rolling_std / rolling_mean
+        # Note: added a non-zero term in denominator to prevent divide by zero errors
+        rolling_std, rolling_mean = moving_sd(a=vmag_filt, w_len=int(fs), skip=1)
+        rolling_cov = rolling_std / (rolling_mean + 1e-12)
 
-        # Detect CoV values about given movement threshold
+        # Detect CoV values above movement threshold
         movement = rolling_cov > self.cov_threshold
 
-        # compile results
+        # Group results
         results = {"movement_detected": movement, "rolling_1s_cov": rolling_cov}
 
         return results
+
+    @staticmethod
+    def _check_input(time, accel, fs=None):
+        """
+        Checks that input meets requirements (see class docstring). Infers fs if necessary.
+
+        Parameters
+        ----------
+        time : array-like
+            Numpy array of unix timestamps. Units of seconds.
+        accel : array-like
+            Numpy array of triaxial accelerometer data.
+        fs : float
+            Sampling rate. Default None. If not provided, will be inferred.
+
+        Returns
+        -------
+        time_ds : array-like
+        accel_ds : array-like
+        fs : float
+            Sampling frequency.
+
+        """
+        # check # of columns
+        _, c = accel.shape
+        if not (c == 3):
+            raise ValueError("Input expected to have 3 columns, but found " + str(c))
+
+        # units must be in G's (mean of magnitude of x,y,z across the entire signal < 4)
+        avg = np.mean(np.linalg.norm(accel, axis=1))
```

Comment:
this is potentially a very expensive check, can this be simplified to check the first minute or something of the recording?

Reply:
Removed